src/app-layer-expectation.c

```diff
@@ -68,13 +68,16 @@ static int g_expectation_data_id = -1;
 SC_ATOMIC_DECLARE(uint32_t, expectation_count);
 
 #define EXPECTATION_TIMEOUT 30
+#define EXPECTATION_MAX_LEVEL 10
 
 typedef struct Expectation_ {
     struct timeval ts;
     Port sp;
     Port dp;
     AppProto alproto;
     int direction;
+    uint8_t level;
+    Flow *orig_f;
```

Comment:
Can we do this in a way that doesn't invite ppl to actually use this flow pointer (by use I mean deref it). I understand you only use it to compare flow addresses now, but this looks like an accident waiting to happen. Flow logic is tricky enough as it is.